xls/modules/zstd/zstd_dec_test.cc

```diff
@@ -0,0 +1,241 @@
+    size_t decompressed_size =
+        ZSTD_getFrameContentSize(frame.data(), frame.size());
+    EXPECT_FALSE(ZSTD_isError(decompressed_size));
+
+    // Decompress the frame with libzstd
+    uint8_t* lib_decomp = new uint8_t[decompressed_size];
+    size_t lib_decomp_size = ZSTD_decompress(lib_decomp, decompressed_size,
+                                             frame.data(), frame.size());
+    EXPECT_FALSE(ZSTD_isError(lib_decomp_size));
+
+    std::vector<uint8_t> sim_decomp;
+    size_t sim_decomp_size_words =
+        (decompressed_size + sizeof(uint64_t) - 1) / sizeof(uint64_t);
+    size_t sim_decomp_size_bytes =
+        (decompressed_size + sizeof(uint64_t) - 1) * sizeof(uint64_t);
+    sim_decomp.reserve(sim_decomp_size_bytes);
+
+    // Send compressed frame to decoder simulation
+    for (int i = 0; i < frame.size(); i += 8) {
+      auto span = absl::MakeSpan(frame.data() + i, 8);
+      auto value = Value(Bits::FromBytes(span, 64));
+      XLS_EXPECT_OK(this->send_queue->Write(value));
+      XLS_EXPECT_OK(this->interpreter->Tick());
+    }
+    PrintTraceMessages("__zstd_dec__ZstdDecoder_0_next");
+
+    // Tick decoder simulation until we get expected amount of output data
+    // batches on output channel queue
+    std::optional<int64_t> ticks_timeout = std::nullopt;
+    absl::flat_hash_map<Channel*, int64_t> output_counts = {
+        {this->recv_queue->channel(), sim_decomp_size_words}};
+    XLS_EXPECT_OK(
+        this->interpreter->TickUntilOutput(output_counts, ticks_timeout));
+
+    // Read decompressed data from output channel queue
+    for (int i = 0; i < sim_decomp_size_words; i++) {
+      auto read_value = this->recv_queue->Read();
+      EXPECT_EQ(read_value.has_value(), true);
+      auto packet =
+          ZstdDecodedPacket::MakeZstdDecodedPacket(read_value.value());
+      EXPECT_EQ(packet.has_value(), true);
+      auto word_vec = packet->GetData();
+      auto valid_length = packet->GetLength() / CHAR_BIT;
+      std::copy(begin(word_vec), begin(word_vec) + valid_length,
+                back_inserter(sim_decomp));
+    }
+
+    EXPECT_EQ(lib_decomp_size, sim_decomp.size());
+    for (int i = 0; i < lib_decomp_size; i++) {
+      EXPECT_EQ(lib_decomp[i], sim_decomp[i]);
+    }
+  }
+};
+
+/* TESTS */
+
+TEST(ZstdLib, Version) { ASSERT_EQ(ZSTD_VERSION_STRING, "1.4.7"); }
+
+TEST_F(ZstdDecoderTest, ParseFrameWithRawBlocks) {
+  int seed = 3;  // Arbitrary seed value for small ZSTD frame
+  auto frame = zstd::GenerateFrame(seed, zstd::BlockType::RAW);
+  EXPECT_TRUE(frame.ok());
+  this->ParseAndCompareWithZstd(frame.value());
+}
+
+TEST_F(ZstdDecoderTest, ParseFrameWithRleBlocks) {
+  int seed = 3;  // Arbitrary seed value for small ZSTD frame
+  auto frame = zstd::GenerateFrame(seed, zstd::BlockType::RLE);
+  EXPECT_TRUE(frame.ok());
+  this->ParseAndCompareWithZstd(frame.value());
+}
+
+//class ZstdDecoderSeededTest : public ZstdDecoderTest,
```

Comment:
any reason this is commented out?

Reply:
A single test case from this test suite generate a random ZSTD frame with `decodecorpus` utility. It is decoded with `libzstd` and then the same encoded frame is processed through the simulation of ZSTD Decoder. The output of the simulation is gathered and compared against results from `libzstd` decoding.
Such test case is repeated 50 times with generating ZSTD frames containing only RAW blocks and 50 times with generating only RLE blocks.
Currently, some of those test cases are failing and we are looking into those. For the time being we commented these tests out so that it would be easily visible if something unexpected will cause the CI to fail in this PR.

Reply:
Not commented out anymore.